tests/llmobs/test_llmobs.py

```diff
@@ -0,0 +1,253 @@
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.METRICS, {"tokens": 100})
+    assert llmobs_events[0]["metrics"] == {"tokens": 100}
+
+
+def test_langchain_span_name_is_set_to_class_name(tracer, llmobs_events):
+    """Test span names for langchain auto-instrumented spans is set correctly."""
+    with tracer.trace(const.LANGCHAIN_APM_SPAN_NAME, resource="expected_name", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+    assert llmobs_events[0]["name"] == "expected_name"
+
+
+def test_error_is_set(tracer, llmobs_events):
+    """Test that error is set on the span event if it is present on the span."""
+    with pytest.raises(ValueError):
+        with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+            llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+            raise ValueError("error")
+    span_event = llmobs_events[0]
+    assert span_event["meta"]["error.message"] == "error"
+    assert "ValueError" in span_event["meta"]["error.type"]
+    assert 'raise ValueError("error")' in span_event["meta"]["error.stack"]
+
+
+def test_model_provider_defaults_to_custom(tracer, llmobs_events):
+    """Test that model provider defaults to "custom" if not provided."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.MODEL_NAME, "model_name")
+    span_event = llmobs_events[0]
+    assert span_event["meta"]["model_name"] == "model_name"
+    assert span_event["meta"]["model_provider"] == "custom"
+
+
+def test_model_not_set_if_not_llm_kind_span(tracer, llmobs_events):
+    """Test that model name and provider not set if non-LLM span."""
+    with tracer.trace("root_workflow_span", span_type=SpanTypes.LLM) as span:
+        span._set_ctx_item(const.SPAN_KIND, "workflow")
+        span._set_ctx_item(const.MODEL_NAME, "model_name")
+    span_event = llmobs_events[0]
+    assert "model_name" not in span_event["meta"]
+    assert "model_provider" not in span_event["meta"]
+
+
+def test_model_and_provider_are_set(tracer, llmobs_events):
+    """Test that model and provider are set on the span event if they are present on the LLM-kind span."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.MODEL_NAME, "model_name")
+        llm_span._set_ctx_item(const.MODEL_PROVIDER, "model_provider")
+    span_event = llmobs_events[0]
+    assert span_event["meta"]["model_name"] == "model_name"
+    assert span_event["meta"]["model_provider"] == "model_provider"
+
+
+@pytest.fixture
+def mock_logs():
+    with mock.patch("ddtrace.llmobs._trace_processor.log") as mock_logs:
+        yield mock_logs
```

Comment:
should we move this up to top of the file to be grouped with the rest of the fixtures?

Reply:
i was planning to do away with it altogether in #11781, but i'm ok to move it up since that's the convention

Reply:
done in [071a675](https://github.com/DataDog/dd-trace-py/pull/11784/commits/071a675ada963d614274303183ca1f002217ddca)